libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -159,13 +126,11 @@ - (id)init {
         if (bundleOAuthCompletionUrl != nil) {
             self.oauthCompletionUrl = bundleOAuthCompletionUrl;
         }
-        _userAccountMap = [[NSMutableDictionary alloc] init];
-        _temporaryUserIdentity = [[SFUserAccountIdentity alloc] initWithUserId:SFUserAccountManagerTemporaryUserAccountUserId orgId:SFUserAccountManagerTemporaryUserAccountOrgId];
-        _anonymousUserIdentity = [[SFUserAccountIdentity alloc] initWithUserId:SFUserAccountManagerAnonymousUserAccountUserId orgId:SFUserAccountManagerAnonymousUserAccountOrgId];
-        _syncQueue = dispatch_queue_create(kSyncQueue, NULL);
-        [self loadAccounts:nil];
-        [self setupAnonymousUser:self.supportsAnonymousUser autocreateAnonymousUser:self.autocreateAnonymousUser];        
-	}
+        [self migrateUserDefaults];
+        accountsLock = [NSRecursiveLock new];
+
+        [self reload];
```

Comment:
Following this path down, the one remaining reservation I have is that file system I/O takes place at the singleton initialization, to load accounts, where it's technically not necessary.  Can we consider fronting the `userAccountMap` property with a getter that lazy loads the accounts?

Other than the general reservation around state change in the initializer, the immediate impact that I see is that it trickles down into the way the persister class has to be initialized as a result, which in turn limits the flexibility of that class (must instantiate with a known initializer, has to be a class-level setter of a class variable, etc.).

If the persister class could be an `SFUserAccountPersister` _instance_, defined through an instance property of the account manager singleton, you could configure the persister class however you wanted, call whatever initializer/setup you needed for the class, etc.